### DISCUSSION TITLE
How To AI (Almost) Anything

### DISCUSSION THREAD

**Discussion by chengjun** [computational-class/css#71] [General]
[MAS.S60 How2AI](https://mit-mi.github.io/how2ai-course/spring2025/)
Schedule
** Exact topics and schedule subject to change, based on student interests and course discussions. **

Date	Topics	Readings
2/4	Week 1 Introduction [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec1%20-%20introduction.pdf) [[video]](https://youtu.be/0MYt0u3CW5I)
Course syllabus and requirements
Introduction to AI and AI research
[Foundations and Trends in Multimodal Machine Learning: Principles, Challenges, and Open Questions](https://arxiv.org/abs/2209.03430)
[Multimodal Machine Learning: A Survey and Taxonomy](https://arxiv.org/abs/1705.09406)
[Representation Learning: A Review and New Perspectives](https://arxiv.org/abs/1206.5538)

2/6	Week 1 Introduction to AI Research [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec1.2%20-%20AI%20research.pdf) [[video]](https://youtu.be/104FX8MYKAM)
Introduction to AI and AI research
Generating ideas, reading and writing papers, AI experimentation

2/11	Week 2 Foundation 1: Data, structure, information [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec2%20-%20data.pdf) [[video]](https://youtu.be/fLybbXyfN28)
Common data modalities
Data collection strategies
Training objectives and generalization
[Machine learning: Trends, Perspectives, and Prospects](https://www.science.org/doi/abs/10.1126/science.aaa8415)
[Representation Learning: A Review and New Perspectives](https://arxiv.org/abs/1206.5538)
[Foundations and Trends in Multimodal Machine Learning: Principles, Challenges, and Open Questions](https://arxiv.org/abs/2209.03430)
[Geometric Deep Learning: Grids, Groups, Graphs, Geodesics, and Gauges](https://arxiv.org/abs/2104.13478)

2/14	Week 2 Foundation 2: Practical AI tools [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/Debugging%20Tips.pdf) [[video]](https://youtu.be/NOSuy0NOe9k)
Getting started with PyTorch
Huggingface packages
Debugging machine learning models
[A Recipe for Training Neural Networks](https://karpathy.github.io/2019/04/25/recipe/)
[Fine-tuning a Code LLM on Custom Code on a single GPU](https://colab.research.google.com/drive/1EDsjYRrAiujUew0GRJ_hyVoNMsSDnlnx?usp=sharing)
[MAS.S60 Pytorch Introduction](https://colab.research.google.com/drive/1SoTu6gvYcLNDqPwNPTWmsSYF-l-UfHPx?usp=sharing)

2/18	Week 3 No class, shifted President's day
2/20	Week 3 Project proposal presentations

2/25	Week 4 Foundation 3: Model architectures [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec3%20-%20models.pdf) [[video]](https://youtu.be/V0gRkmu4mFY)
Structure and invariances
Temporal sequence models
Spatial convolution models
Models for sets and graphs
[Geometric Deep Learning: Grids, Groups, Graphs, Geodesics, and Gauges](https://arxiv.org/abs/2104.13478)
[An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929)
[Attention Is All You Need](https://arxiv.org/abs/1706.03762)
[Neural Machine Translation by Jointly Learning to Align and Translate](https://arxiv.org/abs/1409.0473)
[Deep Sets](https://arxiv.org/abs/1703.06114)
[Graph Attention Networks](https://arxiv.org/abs/1710.10903)

2/25	Week 4 Discussion 1: Learning and generalization
[Learning the Bitter Lesson](https://arxiv.org/pdf/2410.09649)
[Unifying Grokking and Double Descent](https://arxiv.org/pdf/2303.06173)
[Generalization in Neural Networks](https://arxiv.org/pdf/2209.01610)
[Textbooks are all you Need](https://arxiv.org/pdf/2306.11644)
[A Conceptual Pipeline for Machine Learning](https://arxiv.org/pdf/2207.07528)

3/4	Week 5 Multimodal 1: Connections and alignment [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec4%20-%20multimodal.pdf) [[video]](https://youtu.be/kixc1mh55yY)
Heterogeneity, connections, and interactions
Multimodal technical challenges
Alignment and transformers
[Foundations and Trends in Multimodal Machine Learning: Principles, Challenges, and Open Questions](https://arxiv.org/abs/2209.03430)
[What Makes for Good Views for Contrastive Learning?](https://arxiv.org/abs/2005.10243)
[Characterization and classification of semantic image-text relations](https://link.springer.com/article/10.1007/s13735-019-00187-6)
[When and why vision-language models behave like bags-of-words, and what to do about it?](https://arxiv.org/abs/2210.01936)

3/6	Week 5 Discussion 2: Modern AI architectures
[Scaling Laws for Generative Mixed-Modal Models](https://arxiv.org/abs/2301.03728)
[Not All Tokens Are What You Need for Pretraining](https://arxiv.org/abs/2404.07965)
[PaLI: A Jointly-Scaled Multilingual Language-Image Model](https://arxiv.org/abs/2209.06794)
[The Evolution of Multimodal Model Architectures](https://arxiv.org/abs/2405.17927)
[An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929)
[A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545)
[Inductive Representation Learning on Large Graphs](https://arxiv.org/abs/1706.02216)
[Janossy Pooling: Learning Deep Permutation-Invariant Functions for Variable-Size Inputs](https://arxiv.org/abs/1811.01900)

3/11	Week 6 Multimodal 2: Interactions and fusion [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec5%20-%20fusion.pdf) [[video]](https://youtu.be/Hsv1mOIZ1Ag)
Cross-modal interactions
Multimodal fusion
[Ten Myths of Multimodal Interaction](https://dl.acm.org/doi/pdf/10.1145/319382.319398)
[Multimodal interaction: A review](https://www.sciencedirect.com/science/article/pii/S0167865513002584)
[Quantifying & Modeling Multimodal Interactions: An Information Decomposition Framework](https://arxiv.org/abs/2302.12247)
[Does my multimodal model learn cross-modal interactions? It’s harder to tell than you might think!](https://aclanthology.org/2020.emnlp-main.62/)

3/13	Week 6 Discussion 3: Multimodal alignment
[The Platonic Representation Hypothesis](https://arxiv.org/pdf/2405.07987)
[What Makes for Good Views for Contrastive Learning?](https://arxiv.org/abs/2005.10243)
[Understanding the Emergence of Multimodal Representation Alignment](https://arxiv.org/pdf/2502.16282)
[Does equivariance matter at scale?](https://arxiv.org/abs/2410.23179)
[Learning Transferable Visual Models From Natural Language Supervision?](https://arxiv.org/pdf/2103.00020)
[Emerging Properties in Self-Supervised Vision Transformers](https://arxiv.org/pdf/2104.14294)
[Foundations & trends in multimodal machine learning - Principles, challenges, and open questions](https://arxiv.org/abs/2209.03430)

3/18	Week 7 Multimodal 3: Cross-modal transfer [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec6%20-%20crossmodal.pdf) [[video]](https://youtu.be/k4kywHmr3AA)
Cross-modal learning via fusion
Cross-modal learning via alignment
Cross-modal learning via translation
[LLaVA-Med: Training a Large Language-and-Vision Assistant for Biomedicine in One Day](https://arxiv.org/abs/2303.00915v3)
[DreamLLM: Synergistic Multimodal Comprehension and Creation](https://arxiv.org/abs/2309.11499)
[PaLM-E: An Embodied Multimodal Language Model](https://arxiv.org/abs/2303.03378)

3/20	Week 7 Discussion 4: Multimodal interactions
[Multimodal interaction: A review](https://www.sciencedirect.com/science/article/pii/S0167865513002584)
[Quantifying & Modeling Multimodal Interactions: An Information Decomposition Framework](https://arxiv.org/abs/2302.12247)
[Does my multimodal model learn cross-modal interactions? It’s harder to tell than you might think!](https://aclanthology.org/2020.emnlp-main.62/)
[Kosmos-2: Grounding Multimodal Large Language Models to the World](https://arxiv.org/abs/2306.14824)
[Chameleon: Mixed-modal early-fusion foundation models](https://arxiv.org/abs/2405.09818)
[MM1: Methods, Analysis and Insights from Multimodal LLM Pre-training](https://link.springer.com/chapter/10.1007/978-3-031-73397-0_18)
[MoMa: Efficient Early-Fusion Pre-training with Mixture of Modality-Aware Experts](https://arxiv.org/pdf/2407.21770)

3/25	Week 8 No class, spring break

4/1	Week 9 Large models 1: Large foundation models [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec7%20-%20large%20models.pdf) [[video]](https://youtu.be/ZFLRDEUSX68)
Pre-training data
Self-supervised learning
Fine-tuning, instructing, alignment
[Training Compute-Optimal Large Language Models](https://arxiv.org/abs/2203.15556)
[Super-NaturalInstructions: Generalization via Declarative Instructions on 1600+ NLP Tasks](https://arxiv.org/abs/2204.07705)
[LoRA: Low-Rank Adaptation of Large Language Models](https://arxiv.org/abs/2106.09685)
[A Visual Guide to Mixture of Experts (MoE)](https://newsletter.maartengrootendorst.com/p/a-visual-guide-to-mixture-of-experts)
[A Visual Guide to Quantization](https://newsletter.maartengrootendorst.com/p/a-visual-guide-to-quantization)
[Improved Baselines with Visual Instruction Tuning](https://arxiv.org/abs/2310.03744)

4/3	Week 9 Project midterm presentations

4/8	Week 10 No class, member's week

4/15	Week 11 Large models 2: Large multimodal models [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec8%20-%20large%20multimodal%20models.pdf) [[video]](https://youtu.be/p_GGsKgGxSo)
Multimodal pre-training
Adapting large language models to multimodal
Multimodal LLMs with generation
[Quantifying & Modeling Multimodal Interactions: An Information Decomposition Framework](https://arxiv.org/abs/2302.12247)
[Multimodal Transformer for Unaligned Multimodal Language Sequences](https://arxiv.org/abs/1906.00295)
[Masked Autoencoders Are Scalable Vision Learners](https://arxiv.org/abs/2111.06377)
[Scaling Laws for Native Multimodal Models Scaling Laws for Native Multimodal Models](https://arxiv.org/abs/2504.07951)
[Transfer between Modalities with MetaQueries](https://arxiv.org/abs/2504.06256)

4/17	Week 11 Discussion 5: Large language models
[LoRA: Low-Rank Adaptation of Large Language Models](https://arxiv.org/abs/2106.09685)
[Gated Linear Attention Transformers with Hardware-Efficient Training](https://arxiv.org/abs/2312.06635)
[Unintended Impacts of LLM Alignment on Global Representation](https://arxiv.org/abs/2402.15018)
[A Visual Guide to Quantization](https://newsletter.maartengrootendorst.com/p/a-visual-guide-to-quantization)
[Scaling Instruction-Finetuned Language Models](https://arxiv.org/abs/2106.09685)

4/22	Week 12 Large models 3: Modern generative models [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec9%20-%20generative%20AI.pdf) [[video]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/)
Diffusion models
Controllable generation
Flow Matching
[Scalable Diffusion Models with Transformers](https://arxiv.org/abs/2212.09748)
[Flow Matching in Latent Space](https://arxiv.org/abs/2307.08698)
[Scaling Rectified Flow Transformers for High-Resolution Image Synthesis](https://arxiv.org/abs/2403.03206)
[Movie Gen: A Cast of Media Foundation Models](https://arxiv.org/abs/2410.13720)

4/24	Week 12 Discussion 6: Large multimodal models
[LLaMA-Adapter: Efficient Fine-tuning of Language Models with Zero-init Attention](https://arxiv.org/pdf/2303.16199)
[Cobra: Extending Mamba to Multi-Modal Large Language Model for Efficient Inference](https://arxiv.org/pdf/2403.14520)
[ModaVerse: Efficiently Transforming Modalities with LLMs](https://arxiv.org/abs/2401.06395)
[Spider: Any-to-Many Multimodal LLM](https://arxiv.org/pdf/2411.09439)
[SPHINX-X: Scaling Data and Parameters for a Family of Multi-modal Large Language Models](https://arxiv.org/pdf/2402.05935)
[How Far Are We to GPT-4V? Closing the Gap to Commercial Multimodal Models with Open-Source Suites](https://arxiv.org/pdf/2404.16821)
[NExT-GPT: Any-to-Any Multimodal LLM](https://arxiv.org/pdf/2309.05519)
[Learning to rebalance multi-modal optimization by adaptively masking subnetworks](https://arxiv.org/pdf/2404.08347)

4/29	Week 13 No class, CHI week
5/1	Week 13 Discussion 7: Generative AI
[Large Language Diffusion Models](https://arxiv.org/pdf/2502.09992)
[Compositional Generative Modeling: A Single Model is Not All You Need](https://arxiv.org/pdf/2402.01103)
[Flow Matching for Generative Modeling](https://arxiv.org/abs/2210.02747)
[Flow Matching Guide and Code](https://arxiv.org/abs/2412.06264)
[FlowMotion: Target-Predictive Conditional Flow Matching for Jitter-Reduced Text-to-Motion Generation](https://arxiv.org/abs/2504.01338)
[MusFlow: Multimodal Music Generation via Conditional Flow Matching](https://arxiv.org/abs/2504.13535)
[Unraveling the Connections Between Flow Matching and Diffusion Probabilistic Models](https://arxiv.org/abs/2311.07625)
[Exploring Diffusion and Flow Matching Under Generator Matching](https://arxiv.org/abs/2412.11024)

5/6	Week 14 Interaction 1: Interactive agents and reasoning [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec10%20-%20interaction.pdf) [[video]](https://youtu.be/3SOjOG2oL1w)
Reinforcement learning
Multi-step reasoning
[Deep reinforcement learning from human preferences](https://arxiv.org/abs/1706.03741)
[Deepseek-r1: Incentivizing reasoning capability in LLMs via reinforcement learning](https://arxiv.org/abs/2501.12948)
[Faulty reward functions in the wild](https://openai.com/index/faulty-reward-functions/)
[Direct preference optimization: Your language model is secretly a reward model](https://arxiv.org/abs/2305.18290)

5/8	Week 14 Project final presentations

5/13	Week 15 Interaction 2: Human AI interaction [[slides]](https://mit-mi.github.io/how2ai-course/spring2025/schedule/lec11%20-%20human.pdf) [[video]](https://youtu.be/NrmLjmXU66M)
Interaction mediums
Human in the loop learning
Safety and reliability
[Interactive Sketchpad: A Multimodal Tutoring System for Collaborative, Visual Problem-Solving](https://arxiv.org/abs/2503.16434)
[VideoWebArena: Evaluating Multimodal Agents on Video Understanding Web Tasks](https://arxiv.org/abs/2410.19100)
[OpenVLA: An Open-Source Vision-Language-Action Model](https://arxiv.org/abs/2406.09246)
[ArtPrompt: ASCII Art-based Jailbreak Attacks against Aligned LLMs](https://arxiv.org/abs/2402.11753)
[Guidelines for Human-AI Interaction](https://dl.acm.org/doi/abs/10.1145/3290605.3300233)
